src/main/java/org/apache/mesos/scheduler/repair/CoreFrameworkState.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.state.StateStoreException;
+
+import java.util.List;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework
+ * as a whole.
+ */
+public interface CoreFrameworkState {
```

Comment:
Are we talking about `stopped` or `failed` tasks here.  I think you mean `failed`.  Working under that assumption, removing persistence of information about a Task is not categorically the right approach for permanent Task failure recovery.  A call against an Interface should be made.  In some cases the implementation of that interface may (as in Kafka's case) determine that removal of information from the StateStore is a valid approach.  In `dcos-commons` however, no assumption may be made regarding how a Service intends to handle the determination that a Task has permanently failed.  Removal of this persistent information will certainly be the wrong thing to do for some services.